course3/homework-string/my_string.cc

```diff
@@ -0,0 +1,64 @@
+#include <iostream>
+#include <cstring>
+#include "my_string.h"
+
+MyString::MyString(){
+	buffer = std::unique_ptr<char[]>(new char[1]);
+	buffer = '\0';
```

Comment:
Here yu are setting the _pointer_ to null, dallocating the 1-char buffer allocated just above.
If you'd compile with warning enabled, compiler should warn you here.